File: C#/프로그래머스/0/181901. 배열 만들기 1/배열 만들기 1.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int n, int k)
    {
        List<int> list = new List<int>();
        
        for(int x=1;x<=n;x++)
        {   
            if(x%k == 0)
            {
                list.Add(x);
            }
        
        }
        int[] answer = list.ToArray();
        return answer;
    }
}

File: C/프로그래머스/0/181901. 배열 만들기 1/배열 만들기 1.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n, int k) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int size = n/k;
    
    int* answer = (int*)malloc(size * sizeof(int));
    
    if(answer == NULL)
    {
        exit(1);
    }
       
    for(int x=0;x<size;x++)
    {
        *(answer+x) = k + (x*k);
    }
    return answer;
}